instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -29,11 +31,15 @@ def process(commands) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, M
             attributes['peer.service'] = config[:peer_service] if config[:peer_service]
             attributes.merge!(OpenTelemetry::Instrumentation::Redis.attributes)
 
+            # Parse commands if we need to derive span attributes from them
+            set_value_size_bytes, parsed_commands = parse_commands(commands) if config[:record_value_size] || (config[:db_statement] != :omit)
```

Comment:
Given the complexity of `parse_commands` with this change, and the requirement to allocate an array for the return value, I think we're better off calculating the size separately. I think this looks something like:

```ruby
value_size_bytes = value_size(commands) if config[:record_value_size]

# ...

def value_size(commands)
  value_size = nil
  commands.map do |command|
    command = command[0] if command.is_a?(Array) && command[0].is_a?(Array)
    return nil if command[0] == :auth
    if SET_VALUE_SIZE_COMMANDS.include?(command[0])
      value_size ||= 0
      value_size += calculate_bytesize(command[-1])
    end
  end
  value_size
end
```

Reply:
Yeah I think you're spot on, here. `parse_commands` has gotten too thorny.